### DISCUSSION TITLE
Allow multicast von public interface

### DISCUSSION THREAD

**Discussion by jens4topsnet** [FoobarOy/foomuuri#28] [Ideas]
Hello,
i would like to use foomuuri on my server. The server speak OSPF and OSPF3 to the upstream router. Foomuuri block all multicast traffic if szone and dzone is not localhost. But i need to allow multicast traffic on my public interface.
My solution (not the best) ist change the nft action at line 1423 and 1425 from drop to accept.

A better way is allow multicast via CONFIG parameter.

Jens Roesiger

**Comment by jens4topsnet** [comment]
Here my working patch:

```
--- foomuuri.orig       2024-01-31 09:46:53.684572589 +0100
+++ foomuuri    2024-01-31 10:03:49.741478875 +0100
@@ -59,6 +59,7 @@
     'priority_offset': '5',
     'dbus_firewalld': 'no',
     'nft_bin': 'nft',
+    'multicast_action': 'drop',

     # Directories and files. Files are relative to state_dir.
     'etc_dir': '/etc/foomuuri',
@@ -1420,9 +1421,9 @@
     if ipv == 4:
         output_cast('broadcast', szone, dzone, rules, ipv)
         if not local_local:
-            out('meta pkttype { broadcast, multicast } drop')
+            out('meta pkttype { broadcast, multicast } '+CONFIG['multicast_action'])
     elif not local_local:
-        out('meta pkttype multicast drop')
+        out('meta pkttype multicast '+CONFIG['multicast_action'])

     # Unicast
     output_cast('unicast', szone, dzone, rules, ipv)

```

The value multicast_action can change via foomuuri {multicast_action accept} in config a file

Jens Roesiger

**Comment by kimheino** [comment]
Hi,

> The server speak OSPF and OSPF3 to the upstream router. Foomuuri block all multicast traffic if szone and dzone is not localhost. But i need to allow multicast traffic on my public interface.

To allow all multicast traffic you can do:

```
localhost-public {
  multicast
  ...
}

public-localhost {
  multicast
  ...
}
```

It would be much better to allow OSPF only, not all. I tried to read RFCs and something like this might work:

```
macro {
  ospf     protocol 89 daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6 ; protocol 89 multicast daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6
}

localhost-public {
  ospf
  ...
}

public-localhost {
  ospf
  ...
}
```

Maybe you could test it and I'll add it default.services.conf -file if working?

You can take a look at `foomuuri list macro | grep multicast` to see similar macros.

**Reply by jens4topsnet** [reply]
This solution is exactly right for me. Then the patch is not needed.
Foomuuri is great. Thank you for the development.

**Reply by jens4topsnet** [reply]
Hi,

i have seen, that you update the default.services.conf. Thanks!
But please add the IPv6 LinkLocal IP fe80:: the the ospf macro. Without this IP,the ospf3 link is not comming up.

Here my working ospf service marco:
`ospf     protocol 89 daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6 fe80:: ; protocol 89 multicast daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6 fe80::`

Best Regards
Jens

**Reply by jens4topsnet** [reply]
Hi,
I have an improvement for the ospf rule: the fe80:: address need a netmask. (fe80::/10)
Here the better macro:
`ospf     protocol 89 daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6 fe80::/10 ; protocol 89 multicast daddr 224.0.0.5 224.0.0.6 ff02::5 ff02::6 fe80::/10
`

--
Jens

**Reply by kimheino** [reply]
Thanks, committed.

**Comment by kimheino** [comment]
> Here my working patch:
...
> The value multicast_action can change via foomuuri {multicast_action accept} in config a file

Thanks for the patch. It looks correct, but unfortunately I  have to reject it. See above answer and my comments to very similar PR https://github.com/FoobarOy/foomuuri/pull/14 

Apparently I have to document this much better in wiki.
